crates/egui_glow/examples/pure_glow.rs

```diff
@@ -17,66 +17,91 @@ impl GlutinWindowContext {
+        use glutin::display::GetGlDisplay;
+        use glutin::display::GlDisplay;
+        use glutin::prelude::GlSurface;
+        use raw_window_handle::HasRawWindowHandle;
+        let winit_window_builder = winit::window::WindowBuilder::new()
+            .with_resizable(true)
+            .with_inner_size(winit::dpi::LogicalSize {
+                width: 800.0,
+                height: 600.0,
+            })
+            .with_title("egui_glow example") // Keep hidden until we've painted something. See https://github.com/emilk/egui/pull/2279
+            .with_visible(false);
+
+        let config_template_builder = glutin::config::ConfigTemplateBuilder::new()
             .prefer_hardware_accelerated(None)
             .with_depth_size(0)
             .with_stencil_size(0)
-            .with_transparency(false)
-            .compatible_with_native_window(raw_window_handle)
-            .build();
-
-        let config = gl_display
-            .find_configs(config_template)
-            .unwrap()
-            .next()
-            .unwrap();
+            .with_transparency(false);
 
+        tracing::debug!("trying to get gl_config");
+        let (mut window, gl_config) =
+            glutin_winit::DisplayBuilder::new() // let glutin-winit helper crate handle the complex parts of opengl context creation
+                .with_preference(glutin_winit::ApiPrefence::FallbackEgl) // https://github.com/emilk/egui/issues/2520#issuecomment-1367841150
+                .with_window_builder(Some(winit_window_builder.clone()))
+                .build(
+                    event_loop,
+                    config_template_builder,
+                    |mut config_iterator| {
+                        config_iterator.next().expect(
+                            "failed to find a matching configuration for creating glutin config",
+                        )
+                    },
+                )
+                .expect("failed to create gl_config");
+        let gl_display = gl_config.display();
+        tracing::debug!("found gl_config: {:?}", &gl_config);
+
+        let raw_window_handle = window.as_ref().map(|w| w.raw_window_handle());
+        tracing::debug!("raw window handle: {:?}", raw_window_handle);
         let context_attributes =
-            glutin::context::ContextAttributesBuilder::new().build(Some(raw_window_handle));
-        // for surface creation.
-        let (width, height): (u32, u32) = winit_window.inner_size().into();
+            glutin::context::ContextAttributesBuilder::new().build(raw_window_handle);
+        // by default, glutin will try to create a core opengl context. but, if it is not available, try to create a gl-es context using this fallback attributes
+        let fallback_context_attributes = glutin::context::ContextAttributesBuilder::new()
+            .with_context_api(glutin::context::ContextApi::Gles(None))
+            .build(raw_window_handle);
+        let not_current_gl_context = unsafe {
```

Comment:
This is weirdly named. Usually you don't name things after what they are _not_.

Reply:
comes directly from glutin https://docs.rs/glutin/latest/glutin/context/enum.NotCurrentContext.html